### PR TITLE
Add additional S3 settings for cloning

### PR DESCRIPTION
**Pull request by TeutoNet**
In order to enable cloning from non AWS S3 service additional settings are required, see #494 .

### CODE REVIEW

**Comment by TeutoNet** [comment]
Is it possible that the CI is broken, because the upstream project moved from `zalando-incubator` to `zalando`?

File: docs/user.md [hunk 1]
```diff
@@ -242,6 +242,25 @@ metadata:
...
+spec:
+  clone:
+    uid: "efd12e58-5786-11e8-b5a7-06148230260c"
+    cluster: "acid-batman"
+    timestamp: "2017-12-19T12:40:33+01:00"
+    s3_endpoint: https://s3.acme.org
+    s3_access_key_id: 0123456789abcdef0123456789abcdef
+    s3_secret_access_key: 0123456789abcdef0123456789abcdef
+    s3_force_path_style: true
+    aws_instance_profile: "0"
```
Comment (CyberDem0n) [inline docs/user.md]: You don't need to propagate aws_instance_profile, it is automatically calculated in spilo based on the presence of s3_access_key_id and s3_secret_access_key

Comment (TeutoNet) [reply docs/user.md]: Ok, should I remove the setting at all?

**Comment by erthalion** [comment]
> Is it possible that the CI is broken, because the upstream project moved from `zalando-incubator` to `zalando`?

Most likely, we'll fix it shortly.

File: docs/user.md (hunk 1 above)
Comment (TeutoNet) [reply docs/user.md]: @CyberDem0n  I've removed the mentioned `aws_instance_profile`.

**Comment by Jan-M** [comment]
@sdudoladov @CyberDem0n how to progress here? Is there anything in Spilo will make it simpler?

**Comment by CyberDem0n** [comment]
There are a [bunch](https://github.com/zalando/spilo/blob/master/postgres-appliance/scripts/configure_spilo.py#L595-L607) of env variables is Spilo to support backup and PITR with wal-e/wal-g.
Most of them could also be specified with `CLONE_` or `STANDBY_` prefix.

The list of settings might look huge, but unfortunately we can't make it smaller and I am pretty sure there are some settings already missing due to some changes in wal-g.

**Comment by abh** [comment]
We don't use AWS S3, but have a compatible blob store -- this change seems to do exactly what we need...

**Comment by TeutoNet** [comment]
I've updated my PR branch to include latest changes from master.

**Comment by Jan-M** [comment]
@FxKu @sdudoladov looks good to me, if this helps with non aws non s3 setups this needs to be exposed.

**Comment by sdudoladov** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Comment by FxKu** [comment]
Thanks @TeutoNet for your contribution.

**Pull request merged**